db/db_impl/db_impl_open.cc

```diff
@@ -144,6 +145,23 @@ DBOptions SanitizeOptions(const std::string& dbname, const DBOptions& src,
     result.wal_dir = result.wal_dir.substr(0, result.wal_dir.size() - 1);
   }
 
+#ifdef OS_LINUX
```

Comment:
Could this be done in https://github.com/facebook/rocksdb/blob/main/include/rocksdb/file_system.h#L659? This logic is only applicable to PosixEnv.

Reply:
Good point - moved it to PosixFileSystem